File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.java
// Last updated: 3/14/2026, 11:51:00 AM
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            set.add(nums[i]);
        }
        int largest = 0;
        for (Integer i : set) {
            int count = 1;
            if(!set.contains(i-1)){
                while(true){
                    if(set.contains(i +1 )){
                        count++;
                        i++;
                    }
                    else {
                        break;
                    }
                }
            }
            largest = Math.max(largest,count);
        }
        return largest;
    }
}

File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.py
# Last updated: 3/15/2026, 11:37:15 AM
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        setArray= set()
        count = 1
        for i in nums:
            setArray.add(i)
        for nums in setArray:
            if nums - 1 in setArray:
                continue
            else:
                tempCount = 1
                while True:
                    if nums + 1 in setArray:
                        tempCount+=1
                        nums+=1
                    else:
                        break
                if count < tempCount:
                    count = tempCount


        return count